dev/CommandBarFlyout/CommandBarFlyout.cpp

```diff
@@ -252,26 +252,135 @@ winrt::Control CommandBarFlyout::CreatePresenter()
     presenter.BorderThickness(winrt::ThicknessHelper::FromUniformLength(0));
     presenter.Padding(winrt::ThicknessHelper::FromUniformLength(0));
     presenter.Content(*commandBar);
-    // Clear the default CornerRaius(4) on FlyoutPresenter, CommandBarFlyout will do its own handling.
-    if (winrt::IControl7 presenterControl7 = presenter)
+
+    // Disable the default shadow, as we'll be providing our own shadow.
+    if (winrt::IFlyoutPresenter2 presenter2 = presenter)
     {
-        presenterControl7.CornerRadius({ 0 });
+        presenter2.IsDefaultShadowEnabled(false);
     }
 
-    if (!SharedHelpers::Is21H1OrHigher())
+    auto cornerRadiusResource = ResourceAccessor::ResourceLookup(*commandBar, box_value(c_controlCornerRadiusKey));
+    winrt::CornerRadius primaryCornerRadius{ 4,4,4,4 };
+    if (cornerRadiusResource)
     {
-        // This logic applies to projected shadows, which are the default on < 21H1.
-        // When on 21H1 or higher, drop shadows are the default and need to be applied higher up
-        // in the tree to avoid being clipped away.  The default shadow works great for this.
-        // For < 21H1, we will provide our own shadow, not the one that FlyoutPresenter has by default.
-        // We need to specifically target the CommandBar for the shadow, not the default node far
-        // above that.
-        if (winrt::IFlyoutPresenter2 presenter2 = presenter)
+        primaryCornerRadius = unbox_value<winrt::CornerRadius>(cornerRadiusResource);
+    }
+
+    if (winrt::IControl7 presenterControl7 = presenter)
+    {
+        // When >21H1, we'll need to manage the presenter's CornerRadius when the overflow is opened/
+        // closed. In order for drop shadows to render correctly, the element with the shadow needs to have
+        // the correct corner radius as well. Otherwise the shadow will render with sharp corners.
+        if (SharedHelpers::Is21H1OrHigher())
+        {
+            presenterControl7.CornerRadius(primaryCornerRadius);
```

Comment:
>CornerRadius [](start = 30, length = 12)

Can we just set the corner radius on all versions?

Reply:
Actually, yes we can! Tested it on 20H2 and 21H1.